### PR TITLE
[BIOMAGE-2255] - Test the subset pipeline

### PR DESCRIPTION
**Pull request by gerbeldo**
# Description
Add unittest and snaptshot test to the subset pipeline.

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-2255

#### Link to staging deployment URL (or set N/A)
N/A

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: pipeline-runner/tests/testthat/test-subset-1-subset_seurat.R [hunk 1]
```diff
@@ -1,36 +1,138 @@
...
-# input <- mock_input()
+library(Seurat)
+
+mock_parent_experiment_data <- function(parent_experiment_id) {
+  paths <- setup_test_paths()
+  qc_snaps_path <- file.path(paths$snaps, "qc")
+
+  parent_scdata_name <-
+    paste0(paste(parent_experiment_id, "integrated_scdata", sep = "-"),
+           ".R")
```
Comment (saracastel) [inline pipeline-runner/tests/testthat/test-subset-1-subset_seurat.R:9]: why double `paste` and not directly `paste0(parent_experiment_id, "-", "integrated_scdata", ".R")`?

File: pipeline-runner/tests/testthat/test-subset-1-subset_seurat.R [hunk 2]
```diff
@@ -1,36 +1,138 @@
...
+  list(
+    parentExperimentId = parent_experiment_id,
+    experimentId = "mock_subset_experiment_id",
+    cellSetKeys = cellset_keys
+  )
+}
+
+
+stub_load_parent_experiment_data <- function(input, pipeline_config) {
+  mock_parent_experiment_data(input$parentExperimentId)
```
Comment (saracastel) [inline pipeline-runner/tests/testthat/test-subset-1-subset_seurat.R:43]: Couldn't `mock_parent_experiment_data` be called directly [here](https://github.com/biomage-org/pipeline/blob/a1a42eb3d91aee19b4ad9f579d81233342266669/pipeline-runner/tests/testthat/test-subset-1-subset_seurat.R#L51) given that this function is calling only `mock_parent_experiment_data`?

File: pipeline-runner/tests/testthat/test-subset-1-subset_seurat.R (hunk 1 above)
Comment (gerbeldo) [reply pipeline-runner/tests/testthat/test-subset-1-subset_seurat.R:9]: sure, I'll change it.

File: pipeline-runner/tests/testthat/test-subset-1-subset_seurat.R (hunk 2 above)
Comment (gerbeldo) [reply pipeline-runner/tests/testthat/test-subset-1-subset_seurat.R:43]: I think it's not as clean but sure. I'll add an empty pipeline config parameter to the mock_parent_experiment_data function (the stubbing function needs to have the same arguments as the function to stub)

**Review by saracastel — APPROVED** [review]

**Pull request merged**